torch/csrc/jit/passes/reconstruct_scopes.cpp

```diff
@@ -0,0 +1,105 @@
+#include <torch/csrc/jit/passes/reconstruct_scopes.h>
+#include <torch/csrc/jit/jit_log.h>
+
+namespace torch {
+namespace jit {
+
+class ReconstructScopesPass {
+ public:
+  ReconstructScopesPass(const Module& m, Graph& g, std::string p)
+      : root_module(&m), graph(&g), prefix(std::move(p)) {};
+  void run();
+
+ private:
+  const Module* root_module;
+  Graph* graph;
```

Comment:
Is there a reason why we are storing Module and Graph as pointers? Can we instead store const refs? Like `const Module& root_module`?

Reply:
I guess we update graph nodes. Is that why it is not const?

Reply:
Actually I missed something. We dont want to store pointers/references to something that may have shorter lifetime than the object itself. If we construct ReconstructScopesPass and pass it around while the module/graph that was passed in is destroyed, we are holding onto something that is not valid.
Either we ensure that they remain valid through the lifetime of ReconstructScopesPass, or we should just copy.

Reply:
OK, I understand. We do need original module as we are mapping Function pointers.